contracts/ERC20Safe.sol

```diff
@@ -244,23 +255,29 @@ contract ERC20Safe is Initializable, BridgeRole, Pausable {
       @notice Deposit initial supply for an ESDT token already deployed on MultiversX
       @param tokenAddress Address of the contract for the ERC20 token that will be deposited
       @param amount number of tokens that need to be deposited
-\   */
-    function initSupply(address tokenAddress, uint256 amount) external onlyAdmin {
+    */
+    function initSupply(address tokenAddress, uint256 amount) public onlyAdmin {
         require(whitelistedTokens[tokenAddress], "Unsupported token");
+        require(!_isTokenMintBurn(tokenAddress), "Cannot init for mintable/burnable tokens");
+        require(nativeTokens[tokenAddress], "Only native tokens can be stored!");
 
-        if (!_isTokenMintBurn(tokenAddress)) {
-            totalBalances[tokenAddress] += amount;
-            IERC20 erc20 = IERC20(tokenAddress);
-            erc20.safeTransferFrom(msg.sender, address(this), amount);
-            return;
-        }
+        totalBalances[tokenAddress] += amount;
```

Comment:
I think this function can be called multiple times and it doesn't generate a cross chain message, so adding `amount` to the current value of `totalBalances[tokenAddress]` should be fine.